android/src/main/java/com/sunmiexternalprinter/TcpIpOutputStream.kt

```
@@ -104,22 +126,56 @@ class TcpIpOutputStream @JvmOverloads constructor(host: String?, port: Int = 910
+   *
+   * @param timeoutMs Maximum time to wait for the background thread (default: 30000ms)
    */
-  fun closeSocket(promise: Promise){
+  fun shutdown(timeoutMs: Long = 30000L) {
     try {
-      // For TCP IP only need to close it once
+      // Close the piped output stream to signal EOF to the background thread.
+      // This causes pipedInputStream.read() to return -1, ending the write loop.
+      try {
+        this.close()
+      } catch (e: IOException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)